### ISSUE TITLE
How to use the result of task after it's completed?

### ISSUE THREAD

**Issue by simkimsia** [coleifer/huey#660]
I have read #365  but not sure how to use it for my own situation.

So I am using huey with Django 3.2

I have the following tasks.py

```python
from core.models import HueyJob
from huey.contrib.djhuey import signal, task
from core.utils import call_third_party_api

call_third_party_api_task = task()(call_third_party_api)


@signal()
def all_signal_handler(signal, task, exc=None):
    status = signal.upper()
    job_id = task.id

    if settings.DEBUG:
        print(f"{signal} - {task.id}")

    HueyJob.objects.update_or_create(
        job_id=job_id,
        defaults={"status": status},
    )
```

and core/models.py

```python
class HueyJob(models.Model):
    STATUS_CANCELED = "CANCELED"
    STATUS_COMPLETE = "COMPLETE"
    STATUS_ERROR = "ERROR"
    STATUS_EXECUTING = "EXECUTING"
    STATUS_EXPIRED = "EXPIRED"
    STATUS_LOCKED = "LOCKED"
    STATUS_RETRYING = "RETRYING"
    STATUS_REVOKED = "REVOKED"
    STATUS_SCHEDULED = "SCHEDULED"
    STATUS_INTERRUPTED = "INTERRUPTED"

    STATUS = (
        (STATUS_CANCELED, "CANCELED"),
        (STATUS_COMPLETE, "COMPLETE"),
        (STATUS_ERROR, "ERROR"),
        (STATUS_EXECUTING, "EXECUTING"),
        (STATUS_EXPIRED, "EXPIRED"),
        (STATUS_LOCKED, "LOCKED"),
        (STATUS_RETRYING, "RETRYING"),
        (STATUS_REVOKED, "REVOKED"),
        (STATUS_SCHEDULED, "SCHEDULED"),
        (STATUS_INTERRUPTED, "INTERRUPTED"),
    )

    status = models.CharField(
        max_length=12,
        choices=STATUS,
        default=STATUS_SCHEDULED,
    )

    job_id = models.UUIDField(
        primary_key=True,
        editable=False
    )
```
and utils.py

```python
def call_third_party_api():
    """Call Thrd partt API and keep polling until success

    :return: thing
    :rtype: thing
    """
    
    return thing
```

the thing that's returned is an object that i want to use

over at views.py,  i would like to also update `native_server.provider_id = the_thing_after_call_third_party_api_is_completed.id`

but I am not sure how.

```python
   task = call_third_party_api_task()

    job, _ = HueyJob.objects.get_or_create(
        job_id=task.id,
        defaults={'status': HueyJob.STATUS_SCHEDULED},
    )

    native_server.create_job = job
    # i would like to also update native_server.provider_id = the_thing_after_call_third_party_api_is_completed.id
    native_server.save()

    return JsonResponse({ 'jobId': task.id }, status=201)
```

I guess i can also execute the query inside the task as outlined here

https://huey.readthedocs.io/en/1.11.0/contrib.html#tasks-that-execute-queries

but i also want to associate native_server with my task.id which I can only do outside the task. So I was wondering if you have any good advice?

### Summary

In short, I have a Django model object that's already created called native_server. I will use huey to do a long running task that will go interact with a third party API. 

I want to save both the task.id and the result of the task with this model object. I don't know how to do this exactly.

**Comment by simkimsia** [comment]
Thank you @coleifer  in the end I did use your option 1 as I don't quite understand the other options.

but i needed to change the task to db_task for option1

Thank you all the same.
